lib/redcord/tracer.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# typed: strict
+
+module Redcord::Tracer
```

Comment:
I guess I find it a bit strange for this logic to be included into the various Redcord models, instead of being a "singleton" somewhere else that gets called into. Likely doesn't matter much but theoretically that would make it harder for someone to extend and mess with this in ways we wouldn't want.

Reply:
currently, it's only mixed in Redcord::Base. it's using a `tracer` attribute that comes from the top-level configuration.